trapper/metrics/metric_handlers/metric_handler.py

```diff
@@ -0,0 +1,119 @@
+import logging
+from typing import Optional, Tuple, Union
+
+import datasets
+import numpy as np
+
+from trapper.common import Registrable
+from trapper.data import IndexedInstance
+from trapper.data.label_mapper import LabelMapper
+from trapper.data.tokenizers import TokenizerWrapper
+
+logger = logging.getLogger(__file__)
+
+
+class MetricHandler(Registrable):
```

Comment:
We could implement the postprocessing in another class I guess. The two tasks seems unrelated and they do not use / manipulate common instance variables, are they?

Reply:
They do if required, `postprocessing()` uses attributes stored by `extract_metadata()` e.g in QA case.

Reply:
I don't see how. Btw, QA does not even override the `postprocess`, it uses the base method which returns the score without a change.

Reply:
> I don't see how. Btw, QA does not even override the `postprocess()`, it uses the base method which returns the score without a change.

`preprocess` method was named `postprocess` before the changes, it was named `postprocess` because it postprocesses the predictions outputs, but now `postprocess` named `preprocess` as it preprocesses the prediction outputs before metric computation, and current `postprocess` method is for score manipulation (I know it's a bit confusing after the change :sweat_smile: )

Reply:
I am talking about preprocess vs postprocess, and still not understood how these two methods are expected to operate on the same instance attributes.